src/dask_histogram/boost.py

```diff
@@ -242,12 +190,73 @@ def dask_name(self) -> str:
 
     @property
     def dask(self) -> HighLevelGraph:
+        if self._dask_name == "__not_yet_calculated__" and self._dask is None:
+            self._build_taskgraph()
         if self._dask is None:
             raise RuntimeError(
                 "The dask graph should never be None when it's requested."
             )
         return self._dask
 
+    def _build_taskgraph(self):
+        data_list = []
+        weights = []
+        samples = []
+
+        dask_data = tuple(
+            datum
+            for datum in (
+                self._staged[0]["args"] + tuple(self._staged[0]["kwargs"].values())
+            )
+            if is_dask_collection(datum)
+        )
+
+        if is_dask_awkward_like(dask_data[0]):
+
```

Comment:
This being such a big if, comments describing the two arm definitely helpful.
The first case if where (multiple) dak arrays have been used to fill the hist. How come we only look at the very first dask collection?

Reply:
this is removed now and dealt with more appropriately